### DISCUSSION TITLE
Linux/Power (64-bit), Linux/RISC-V (64-bit), Linux/s390x and Linux/LoongArch (64-bit) release assets

### DISCUSSION THREAD

**Discussion by benz0li** [coder/code-server#7074] [Show and tell]
*Unofficial* and *untested* Linux/Power (64-bit, `linux/ppc64le`), Linux/RISC-V (64-bit, `linux/riscv64`), Linux/s390x (`linux/s390x`; for versions ≥ 4.99.2) and Linux/LoongArch (64-bit, `linux/loong64`; for versions ≥ 4.104.0) release assets are available at https://gitlab.b-data.ch/coder/code-server/-/releases.

> [!NOTE]
> 2025-11-20: The release of further versions is paused for the time being.

Available versions:
- [v4.105.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.105.1)
- [v4.105.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.105.0)
- [v4.104.3](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.104.3)
- [v4.104.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.104.2)
- [v4.104.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.104.1)
- [v4.104.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.104.0)
- [v4.103.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.103.2)
- [v4.103.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.103.1)
- [v4.103.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.103.0)
- [v4.102.3](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.102.3)
- [v4.102.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.102.2)
- [v4.102.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.102.1)
- [v4.102.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.102.0)
- [v4.101.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.101.2)
- [v4.101.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.101.1)
- [v4.101.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.101.0)
- [v4.100.3](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.100.3)
- [v4.100.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.100.2)
- [v4.100.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.100.1)
- [v4.100.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.100.0)
- [v4.99.4](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.99.4)
- [v4.99.3](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.99.3)
- [v4.99.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.99.2)
- [v4.99.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.99.1)
- [v4.99.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.99.0)
- [v4.98.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.98.2)
- [v4.98.0](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.98.0)
- [v4.97.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.97.2)
- [v4.96.4](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.96.4)
- [v4.96.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.96.2)
- [v4.96.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.96.1)
- [v4.95.3](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.95.3)
- [v4.95.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.95.2)
- [v4.95.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.95.1)
- [v4.93.1](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.93.1)
- [v4.92.2](https://gitlab.b-data.ch/coder/code-server/-/releases/v4.92.2)

**Comment by dmholtz** [comment]
@benz0li Awesome work 🚀 🚀 🚀  and thank you very much for providing these assets.

Could you please share some insights about how these assets can be built? I would be particularly interested in seeing build scripts for s390x and check if they can be used to build assets for the more recent versions of code-server.

**Reply by benz0li** [reply]
These builds were created using the multi-arch docker manifest [`glcr.b-data.ch/coder/code-server-builder`](https://gitlab.b-data.ch/coder/code-server-builder/container_registry).

Cross reference:

* https://github.com/coder/code-server/issues/6986#issuecomment-2335104284

**Reply by benz0li** [reply]
Patches in place: https://gitlab.b-data.ch/coder/code-server-builder/-/tree/5fbf8a93865d4d9689391276af2f3f67392a3bba/patches

The patch for s390x is the same one they use for [VSCodium](https://github.com/VSCodium/vscodium): https://github.com/VSCodium/vscodium/blob/12b08023848522f884b04173ac0894849dd9fa05/patches/linux/reh/s390x/00-build-override-dependency.patch

I eventually gave up on the patches for [Code - OSS](https://github.com/microsoft/vscode) itself:

* https://github.com/VSCodium/vscodium/blob/578df58363326848eebd1074b8bd029fc0c2060b/patches/linux/fix-dependencies.patch

Which seem necessary because of

* https://github.com/microsoft/vscode-vsce/issues/1207
* https://github.com/microsoft/vscode/issues/282188

**Reply by dmholtz** [reply]
Thank you very much for your prompt response and sharing these insights and the patches. Really helpful!
